### ISSUE TITLE
Pubs download results in image file

### ISSUE THREAD

**Issue by cbrody** [IIED-org/IIED-main#456]
There is an issue with some pubs in that when you click to download, it takes you to a jpg of the cover rather than downloading the doc.

Examples:
-	https://www.iied.org/16674iied 
-	https://www.iied.org/10209iied
-	https://www.iied.org/17709iied

Not sure how widespread it is.

**Comment by cbrody** [comment]
Fixed the ones listed above by deleting the media items and re-uploading from pubs-dev.ac.iied.org. For reference I added WP_15 (LIFE-AR) to https://www.iied.org/10209iied and WP_14 (Working with the LDCs...) to https://www.iied.org/17709iied

**Comment by mattjobob** [comment]
English version no longer downloadable from the IIED web site: https://www.iied.org/14624iied

Spanish version just reloads page when click on download

**Comment by mattjobob** [comment]
French https://www.iied.org/fr/g03954 and English https://www.iied.org/g03953 not working

**Comment by cbrody** [comment]
A total of 3030 items seem to be affected. Here's a spreadsheet with Pubs document URLs filtered to exclude items where the last three characters are "pdf": https://docs.google.com/spreadsheets/d/13Xip4hYtBNvsqlZrmT8WKLTKHRJD_w0xuit2Vm8LJ_o/edit#gid=0

My first thought is that one (or more) of the D7 migrations overwrote the media references. It might be possible to write a corrective migration from the D9 pubs library currently on pubs-dev - @finnlewis could you look at this please?

**Comment by cbrody** [comment]
http://pubs-dev.ac.iied.org/ has a copy of the pubs site taken prior to the deployment of iied-main on 30/09/2022

**Comment by mattjobob** [comment]
@finnlewis Just to add to Chris' message, ahead of a meeting we have first thing tomorrow, it would be very helpful for our planning to have some kind of update by the end of today please. Thanks

**Comment by finnlewis** [comment]
Hi @cbrody and @mattjobob 
Had a slow start as the original migration module is uninstalled but configuration is still present.
I've created a new module iied_migration_fixes and am working through the queries and events needed to update the media entities. I'll update again when I have more to report. 

Can I check a couple of things: 

1. Are we expecting all the files to be in the same / right place? I would like to work on the assumption we don't need to move any files on the file system. So far those that I've checked seem to be there. 
2. Are we only concerned with document media entities or do we want to look at all files in the source data and perhaps other media entities that have file references that have been overwritten?

**Comment by cbrody** [comment]
Thanks @finnlewis yes uninstalling the modules was perhaps a little premature! The answer to both questions is yes I believe.

**Comment by finnlewis** [comment]
I've got a working migration locally.

 [notice] Processed 7325 items (7325 created, 0 updated, 0 failed, 0 ignored) - done with 'iied_fix_files'

For each item I'm outputting a notice like this: 

` [notice] 1 media entity updated. public://pdfs/2022-05/74366iied_0.pdf The node that references this media entity is /node/74366 or /20956iied . `

This seems to fix all the publication nodes I've randomly selected to check.. 

Note: there are 7325 files and media entities, not all of which are actually referenced from publication nodes. 

Here's the branch: https://github.com/IIED-org/IIED-main/tree/feature/456-fix-files-migration 

You should be able to see from the query how we're getting the sources: 

https://github.com/IIED-org/IIED-main/blob/feature/456-fix-files-migration/docroot/modules/custom/iied_migrate_fixes/src/Plugin/migrate/source/FixFiles.php#L22

Basically, all files referenced from a `field_media_document` field. This field is only used on document media type, so that should be safe enough and cover all the pdfs of immediate concern.

Just had a thought about revisions. I wonder do we need to update revisions of media entities?

**Comment by cbrody** [comment]
Thanks v. much @finnlewis, I don't think we need to be concerned with revisions. Do you have some time tomorrow to go through the process of running it on Acquia?

**Comment by finnlewis** [comment]
Hi @cbrody 

Following our conversation I have done what we discussed.

1. Took fresh backup from the live production site 
2. Enabled the iied_migrate_fixes module and dependencies
3. run `lando drush mim iied_migrate_fixes` and got  `[notice] Processed 7335 items (7335 created, 0 updated, 0 failed, 0 ignored) - done with 'iied_fix_files'`
4. Uninstalled the iied_migrate_fixes module and dependencies
5. Pushed the resulting database upp to Acquia Cloud, Stage / test environment.

I attach a log of the output notices.

[2022-10-05-0841-migrate-fixes-log.txt](https://github.com/IIED-org/IIED-main/files/9713728/2022-10-05-0841-migrate-fixes-log.txt)

**Comment by cbrody** [comment]
Found a few where a cover image is not displaying despite the PDF file being present and correct:

https://iied-main-stage.ac.iied.org/7799iied
https://iied-main-stage.ac.iied.org/7803iied
https://iied-main-stage.ac.iied.org/12500iied

**Comment by cbrody** [comment]
Re https://github.com/IIED-org/IIED-main/issues/456#issuecomment-1268335466 these all look fine locally with a copy of the files from production, so probably OK to ignore.

**Comment by cbrody** [comment]
@mattjobob @georginadi @DavidIIED I'm happy that the migration has worked as expected. It would be good to do a spot check of the rest of the site, but I'm confident that nothing else should have been affected by the process. @finnlewis could you let us know your availability for running the migration for real on a frozen copy of the site please? Thanks!

**Comment by mattjobob** [comment]
Thanks Chris. I haven't found any further issues - and have had a wander around the site checking generally as well

**Comment by finnlewis** [comment]
Hi @cbrody and @mattjobob - I'm available from now until 4pm, then from 5pm-6pm.

**Comment by finnlewis** [comment]
> Found a few where a cover image is not displaying despite the PDF file being present and correct:
> 
> https://iied-main-stage.ac.iied.org/7799iied https://iied-main-stage.ac.iied.org/7803iied https://iied-main-stage.ac.iied.org/12500iied
> 
> I will test this locally as well.

I think these are missing files. The cover image should be at https://iied-main-stage.ac.iied.org/sites/default/files/styles/page/public/pdfs/migrate/7799IIED.pdf-p1.jpeg but the source file is not present.

Maybe copying the files directory from live to test might fix these?

**Comment by mattjobob** [comment]
Solution implented and all fixed
